src/build/env.py

```diff
@@ -293,14 +293,21 @@ def _find_executable_and_scripts(path: str) -> Tuple[str, str, str]:
     """
     config_vars = sysconfig.get_config_vars().copy()  # globally cached, copy before altering it
     config_vars['base'] = path
+    # Python distributors with custom default installation scheme can set a
+    # scheme that can't be used to expand the paths in a venv.
+    # This can happen if build itself is not installed in a venv.
```

Comment:
I don't think this is relevant or that something distributors should be doing, varying the scheme at runtime depending on the prefix.

Reply:
That is the reality. We need to do that. See mostly https://bugs.python.org/issue43976

Reply:
Unfortunately, this was made possible with the introduction of `sysconfig._get_preferred_schemes` in 3.10, so something like this is needed.